src/stest/src/file.rs

```diff
@@ -0,0 +1,232 @@
+use clap::Parser;
+use std::clone::Clone;
+use std::fmt::Error as FmtError;
+use std::fmt::{Display, Formatter};
+use std::fs::Metadata;
+use std::fs::read_dir;
+use std::io::Error as IOError;
+use std::os::unix::fs::{FileTypeExt, PermissionsExt};
+use std::path::Component;
+use std::path::PathBuf;
+use std::result::Result;
+use std::str::FromStr;
+
+/// Wraps a [PathBuf] for use in stest.
+///
+/// These files are assumed to be on a unix system. Error-handling is simple: always panic.
```

Comment:
Just unwrapping makes for very messy errors. In any place you *actually* expect to see an error, you should at least `expect("A better error message")`. Realistically, `expect` isn't as good as just printing an error and exiting, as it also prints source line info (not clean).

Reply:
In the case of this program, I believe the right way of doing it would be to return a `Result<bool, String>` from all of the test functions. `true` or `false` are the test results. `Error` would then be a problem performing the test. For example, `Problem checking hidden status: File has an empty path`.

Reply:
You could also return `Result<bool,std::error::Error>` so that you can mix custom string errors with io errors. `std::error::Error` implements display so you can just print and exit at top level.

Reply:
Finally got a version that folds the `Result` type throughout. :)